src/scripts/directives.js

```diff
@@ -143,10 +143,15 @@
                 },
                 templateUrl: 'templates/type/radio-input.html',
                 link: function (scope) {
-                    scope.localModel = {
-                        value: scope.model
-                    };
-
+                    scope.localModel = '';
+                    var modelWatchDeregister = scope.$watch('model', function (value) {
+                        if (angular.isDefined(value)) {
+                            scope.localModel = {
+                                value: scope.model
+                            };
+                            modelWatchDeregister();
```

Comment:
I agree. However, object copy-by-reference makes it moot. But that isn't immediately visible or intuitive. Also, `scope.model` could be a primitive, like it is in the test.